lib/consumers.js

```diff
@@ -180,18 +180,17 @@ ConsumerStream.prototype.read = function() {
                 for(var i = 0; i < msgs.length; i++) {
                     decoded_msgs.push(this.schemaClass.decodeMessage(msgs[i]));
                 }
-                this.emit('read', decoded_msgs);
```

Comment:
This breaks the current interface. If you would like to also allow listening on the `read` event for backwards compatibility, you will need to bind the `data` event to `read` the first time a listener is added to the `read` event.

Listening to the `data` event automatically unpauses a node stream. E.g.,

````javascript
var ConsumerStream = function(instance, topic, options) {
    /* ... */
    this.compatibility = false;
}

ConsumerStream.prototype.on = function(event) {
    if (!this.compatibility && event === 'read') {
        this.on('data', this.emit.bind(this, 'read'));
        this.compatibility = true;
    }

    return Readable.prototype.on.apply(this, arguments);
};
````